src/errors/either.ts

```diff
@@ -24,3 +24,15 @@ export type DefiniteEither<T, U> = {
   error?: T
   result: U
 }
+
+export const isFailure = <T, U>(e: Either<T, U>): e is Left<T> => {
+  return e.error !== undefined
+}
+
+export const isSuccess = <T, U>(e: Either<T, U>): e is Right<U> => {
+  return e.result !== undefined
+}
+
+export const failure = <T>(error: T): Left<T> => ({ error })
```

Comment:
🟢 Just a suggestion, but totally up to you:

Helpers for `Either` should take an `Either` as their argument. (or be a method on the Either class...).

In the case of `failure` and `success` here, that would mean taking the whole `Either` as an argument... _and possibly returning nothing_.

e.g.:
```ts
type Option<T> = Either<null, T>

const failure = <Left, Right>(either: Either<Left, Right>): Option<Left> => either.error ?? null
const success = <Left, Right>(either: Either<Left, Right>): Option<Right> => either.result ?? null
```

Reply:
these helpers are not for unwrapping, but for wrapping. You are using them to return the value without having to build an Either object manually.

Basically, like this:

```
function doSomething() {
  if (hasFailed === true) {
     return failure(new Error(...))
  }

  return success(actualValue)
}
```

Reply:
O! They are either constructors!